### ISSUE TITLE
Samples crash on Android 10 with VK_ERROR_LAYER_NOT_PRESENT 

### ISSUE THREAD

**Issue by lrhamill-arm** [KhronosGroup/Vulkan-Samples#48]
The samples works fine on my Samsung S8 (Android 9) and Google Pixel (Android 9). However, they fail on Samsung S10e (Android 10) and on Google Pixel (Android 10).

I followed the instructions for building for Android with Gradle [here](https://github.com/KhronosGroup/Vulkan-Samples/blob/master/docs/build.md#android). The samples fail to load on my Android 10 devices because vkCreateInstance() returns VK_ERROR_LAYER_NOT_PRESENT:
```
com.khronos.vulkan_samples E/VulkanSamples: [error] [_triangle/hello_triangle.cpp:234] Detected Vulkan error: VK_ERROR_LAYER_NOT_PRESENT
```
This is true even if you comment out the part of hello_triangle.cpp that defines the layers to be loaded. I stepped into the Android Studio debugger and inspected instance_info, it seemed to have no layers configured, and yet I still got VK_ERROR_LAYER_NOT_PRESENT.

In the logcat output, you can see that validation layers are still being found and loaded from the app's lib/arm64 directory:
```
com.khronos.vulkan_samples D/vulkan: added global layer 'VK_LAYER_LUNARG_parameter_validation' from library '/data/app/com.khronos.vulkan_samples-Vl8zpsGcknvYybfWM9fDnA==/lib/arm64/libVkLayer_parameter_validation.so'
```
(This is just one example, there are other libraries that are also loaded.) I don't know where this library is coming from.

**Comment by SaschaWillems** [comment]
This is probably related to #33. Do the samples work without validation enabled?

**Comment by lrhamill-arm** [comment]
I haven't intentionally enabled validation, which I thought was disabled by default. When I `grep -rni 'VKB_VALIDATION_LAYERS'` in the repo, it seems to be disabled:
```
./build/android_gradle/build-native/cmake/debug/arm64-v8a/CMakeCache.txt:350:VKB_VALIDATION_LAYERS:BOOL=OFF
./build/android_gradle/build-native/cmake/debug/armeabi-v7a/CMakeCache.txt:350:VKB_VALIDATION_LAYERS:BOOL=OFF
./build/android_gradle/build-native/cmake/release/arm64-v8a/CMakeCache.txt:350:VKB_VALIDATION_LAYERS:BOOL=OFF
./build/android_gradle/build-native/cmake/release/armeabi-v7a/CMakeCache.txt:350:VKB_VALIDATION_LAYERS:BOOL=OFF
```
I also checked and the device's `enable_gpu_debug_layers` property is unset.

According to the debugger, `instance_info.enabledLayerCount` is 0 and `instance_info.ppEnabledLayerNames` is NULL at the point that vkCreateInstance() is called and returns VK_ERROR_LAYER_NOT_PRESENT.

However, something is still loading layers from the APK's lib/arm64 directory - see the log above.

**Comment by ghost** [comment]
This should be fixed by #34.

@lrhamill-arm - Can you test with PR34 and see if the resolves your issue?

**Comment by SaschaWillems** [comment]
Did you use any debugging tools on that Android device when that happened?

It's possible to enable layer settings per-app that persist through a reboot, which some debug tools may do. That could explain your problems.

See https://developer.android.com/ndk/guides/graphics/validation-layer#enable-layers-outside-app

If you do the following:
> $ adb shell settings list global | grep gpu

And that lists a layer, it means that setting is active and you need to deactivate it.

**Comment by lrhamill-arm** [comment]
Hi, sorry for the late reply, I have been on holiday the last several weeks.

In my original investigation I made sure to clear the relevant settings, so I don't think that was the issue.

I pulled the master branch (head 35ca3dc72c8145874686a81900dc4f6c98db9e87) and rebuilt the app. It seems to work now. This issue appears to have been fixed.

**Comment by ghost** [comment]
Should have been resolved #24 .
